### DISCUSSION TITLE
LDAP integration with Grafana container

### DISCUSSION THREAD

**Discussion by survive-wustl** [NetApp/harvest#580] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
We use LDAP to authenticate to our Harvest 1.6x server configured via some edits to /etc/grafana/ldap.toml to authenticate. I want to do the same with Harvest 2.0 running in containers

**Describe the solution you'd like**
start the grafana container with a custom  /etc/grafana/ldap.toml file

**Describe alternatives you've considered**

**Additional context**

**Comment by cgrinds** [comment, marked as answer]
hi @survive-wustl you are using the Docker compose workflow right? You are starting with something like this? `docker-compose -f prom-stack.yml -f harvest-compose.yml up -d --remove-orphans`

If so, can you make your edits to the `prom-stack.yml` file? That file controls how Grafana starts and is configured via Docker. This [link](https://lab.rapternet.us/?p=382) describes how to setup LDAP with Grafana and Docker.

**Comment by survive-wustl** [comment]
OK, I read the link & think I understand what needs to happen. That said, I have no idea how to get a copy of my ldap.toml file included in the container. Do I copy the file somewhere? Do I stuff it in the prom-stack.yml file?

**Comment by cgrinds** [comment]
The article mentions 
> The ldap configuration file can be mounted to the Grafana container using a docker volume.

`prom-stack.yml` is defines the volumes that the grafana container mounts. You will want to add a new mount at line 39 https://github.com/NetApp/harvest/blob/0ecb0d964f5dce86a9e14400994e7eaf51e985a0/prom-stack.yml#L39

For example:
```
volumes:
    - grafana_data:/var/lib/grafana
    - ./grafana:/etc/grafana/provisioning # import Harvest dashboards
    - <local path to your ldap.toml>:/etc/grafana/ldap.toml     #<====== add this line updating local path to point to your ldap.toml

```

**Comment by survive-wustl** [comment]
OK, that makes sense. Making that change didn't cause the containter to not start 

Any idea how to get these:

[auth.ldap]
enabled = true 
config_file = /etc/grafana/ldap.toml 

changes into grafana.ini to use LDAP? I assume it has to be somewhere outside the container. How would I merge those into the grafana.ini file the container needs?

**Reply by SMylk** [reply]
Modify the prom-stack.yml as proposed by cgrinds, I did it like this:
```
        volumes:
            - grafana_data:/var/lib/grafana
            - ./grafana:/etc/grafana/provisioning # import Harvest dashboards
            - ./docker/grafana/ldap.toml:/etc/grafana/ldap.toml
```
My ldap config file is at  ./docker/grafana/ldap.toml relative to the stack.yml.

To enable ldap auth pass the environment variables:
```
        environment:
            - GF_AUTH_LDAP_ENABLED=true
            - GF_AUTH_LDAP_CONFIG_FILE=/etc/grafana/ldap.toml
```
Check user lookup at "Server Admin/LDAP" page.
